src/main/java/net/earthcomputer/clientcommands/command/ItemGroupCommand.java

```diff
@@ -0,0 +1,361 @@
+                    .then(literal("icon")
+                        .then(argument("icon", itemStack())
+                            .executes(ctx -> changeIcon(ctx.getSource(), getString(ctx, "group"), getItemStackArgument(ctx, "icon").createStack(1, false)))))
+                    .then(literal("rename")
+                        .then(argument("new", string())
+                            .executes(ctx -> renameGroup(ctx.getSource(), getString(ctx, "group"), getString(ctx, "new")))))))
+            .then(literal("add")
+                .then(argument("group", string())
+                    .then(argument("icon", itemStack())
+                         .executes(ctx -> addGroup(ctx.getSource(), getString(ctx, "group"), getItemStackArgument(ctx, "icon").createStack(1, false))))))
+            .then(literal("remove")
+                .then(argument("group", string())
+                    .suggests((ctx, builder) -> suggestMatching(groups.keySet(), builder))
+                    .executes(ctx -> removeGroup(ctx.getSource(), getString(ctx, "group"))))));
+    }
+
+    private static int addGroup(ServerCommandSource source, String name, ItemStack icon) throws CommandSyntaxException {
+        if (groups.containsKey(name)) {
+            throw ALREADY_EXISTS_EXCEPTION.create(name);
+        }
+
+        try {
+            ItemGroup itemGroup = FabricItemGroupBuilder.create(
+                    new Identifier("clientcommands", name))
+                    .icon(() -> icon)
+                    .build();
+
+            groups.put(name, new Group(itemGroup, icon, new ListTag()));
+            saveFile();
+            sendFeedback("commands.citemgroup.addGroup.success", name);
+        } catch (InvalidIdentifierException e) {
+            throw ILLEGAL_CHARACTER_EXCEPTION.create(name);
+        }
+        return 0;
+    }
+
+    private static int removeGroup(ServerCommandSource source, String name) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        groups.remove(name);
+
+        reloadGroups(12);
+        CreativeInventoryScreenAccessor.setSelectedTab(0);
+        CreativeInventoryScreenAccessor.setFabricCurrentPage(0);
+        saveFile();
+        sendFeedback("commands.citemgroup.removeGroup.success", name);
+        return 0;
+    }
+
+    private static int addStack(ServerCommandSource source, String name, ItemStack itemStack) throws CommandSyntaxException {
+        if (!groups.containsKey(name)) {
+            throw NOT_FOUND_EXCEPTION.create(name);
+        }
+
+        Group group = groups.get(name);
+        ListTag items = group.getItems();
+        ItemGroup itemGroup = group.getItemGroup();
+        items.add(itemStack.toTag(new CompoundTag()));
+
+        reloadGroups(itemGroup.getIndex());
+        saveFile();
+        sendFeedback("commands.citemgroup.addStack.success", itemStack.getItem().getName(), name);
```

Comment:
Is `getName()` the same method that's used to display weapons in death messages?

Reply:
Ohh yeah maybe it will return the name of the item if it has been renamed. Let me check that.

Reply:
Nope, it won't. This will always return the default name of the item.